### DISCUSSION TITLE
Gltchy audio when returning to an input.harbor through switch.

### DISCUSSION THREAD

**Discussion by Russsgithub** [savonet/liquidsoap#1927] [Q&A] [answered]
I have a setup of
```
safe = buffer(mksafe(playlist("./playlist")))

i = interactive.bool("switch", true)

s = input.harbour(max=5., icy=true, "/test")

s = switch(track_sensitive=false, [(i, mksafe(s)), ({true}, safe)])

output.icecast(........)
```

When I switch to the safe playlist I get lots of buffer overrun warnings - as expected.

When I switch back to the harbor input I get the max buffer of clean audio (in this case 5s) then a few seconds of crackle and then smooth audio. I need the buffer as small as possible in this case, but even if i set a 20s mx buffer size I get the buffer size of clean audio followed by a few seconds of crackle.

Possibly related to this .... if I add a crossfade under the switch or any transitions I get a loop of fade in fade out fade in fade out.

**Comment by Russsgithub** [comment]
Using an output.dummy(s) before the switch stopped this.

**Comment by toots** [comment, marked as answer]
That clearly pointed to an issue where `input.harbor` should have been an active operator (which does the same as adding `output.dummy`). @Russsgithub any chance you could confirm the fix and that it works fine now? It's been pushed to both `main` and `v2.0.0`? Would be great to confirm that `input.harbor` still works fine with it. All you gotta do is pull the latest version and remove `output.dummy`. Thanks for the pointer!

**Reply by Russsgithub** [reply]
@toots 
I can confirm that that works now without output.dummy, Thanks.

**Comment by Russsgithub** [comment]
Will do in an hour or two. Thanks

On 17 Sep 2021 14:18, Romain Beauxis ***@***.***> wrote:

That clearly pointed to an issue where input.harbor should have been an active operator (which does the same as adding output.dummy). @Russsgithub<https://github.com/Russsgithub> any chance you could confirm the fix and that it works fine now? It's been pushed to both main and v2.0.0? Would be great to confirm that input.harbor still works fine with it. All you gotta do is pull the latest version and remove output.dummy. Thanks for the pointer!

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/savonet/liquidsoap/discussions/1927#discussioncomment-1347522>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AB627QMKONSQJPFKENP7L2TUCM54BANCNFSM5EAN5JEA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
